### PR TITLE
Add Readme for Analyzers, ObjectPool and Identity

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Part of https://github.com/dotnet/aspnetcore/issues/27561.

For review:

Analyzers and ObjectPool: @pranavkm 
Identity: @HaoK @Tratcher @blowdart

### CODE REVIEW

File: src/Analyzers/Readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## Description
+
+- `Analyzers`: Contains analyzers for ASP.NET Core shipped as part of the Microsoft.NET.Sdk.Web.
+- `Microsoft.AspNetCore.Analyzer.Testing`: Contains types for writing analyzer tests.
+
+## Development Setup
+
+### Build
+
+To build this specific project from source, follow the instructions [on building the project](../../docs/BuildFromSource.md#step-3:-build-the-repo).
```
Comment (JunTaoLuo) [inline src/Analyzers/Readme.md:14]: @captainsafia I noticed the template README.md is now out of date after the change https://github.com/dotnet/aspnetcore/commit/70e7c40a5091e0a01c80aafe416f8f49b62129ed#diff-57a537bf86f9dd51ea61e7b1bacd3adcfe77e7bb051c2245257e9a2fbac06ed8.

**Review by HaoK — APPROVED** [review]

File: src/Analyzers/Readme.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Object Pool
```
Comment (BrennanConroy) [inline src/Analyzers/Readme.md:1]:
```suggestion
# Analyzers
```

File: src/Analyzers/Readme.md (hunk 1 above)
Comment (BrennanConroy) [reply src/Analyzers/Readme.md:14]:
```suggestion
To build this specific project from source, follow the instructions [on building the project](../../docs/BuildFromSource.md#step-3-build-the-repo).
```

File: src/Identity/README.md [hunk 3]
```diff
@@ -1,7 +1,35 @@
-ASP.NET Core Identity
-=====================
+# Identity
 
-ASP.NET Core Identity is the membership system for building ASP.NET Core web applications, including membership, login, and user data. ASP.NET Core Identity allows you to add login features to your application and makes it easy to customize data about the logged in user.
+ASP.NET Core Identity is the membership system for building ASP.NET Core web applications, including membership, login, and user data. ASP.NET Core Identity allows you to add login features to your application and makes it easy to customize data about the logged in user. You can find additional information in the [ASP.NET Core Documentation](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/identity).
```
Comment (BrennanConroy) [inline src/Identity/README.md:3]:
```suggestion
ASP.NET Core Identity is the membership system for building ASP.NET Core web applications, including membership, login, and user data. ASP.NET Core Identity allows you to add login features to your application and makes it easy to customize data about the logged in user. You can find additional information in the [ASP.NET Core Documentation](https://docs.microsoft.com/aspnet/core/security/authentication/identity).
```

File: src/Identity/README.md [hunk 4]
```diff
@@ -1,7 +1,35 @@
...
+- `Specification.Tests`: Contains a test suite for ASP.NET Core Identity store implemenations.
+- `test`: Contains the unit and functional tests for Microsoft.Extensions.Identity and Microsoft.AspNetCore.Identity components.
+- `testassets`: Contains a webapp used for functional testing.
+- `UI`: Contains compiled Razor UI components for use in ASP.NET Core Identity.
+
+## Development Setup
+
+### Build
+
+To build this specific project from source, follow the instructions [on building the project](../../docs/BuildFromSource.md#step-3:-build-the-repo).
```
Comment (BrennanConroy) [inline src/Identity/README.md:24]:
```suggestion
To build this specific project from source, follow the instructions [on building the project](../../docs/BuildFromSource.md#step-3-build-the-repo).
```

File: src/ObjectPool/Readme.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Object Pool
+
+This directory contains sources for [`Microsoft.Extensions.ObjectPool`](https://www.nuget.org/packages/Microsoft.Extensions.ObjectPool). This package provides types that enable object reuse. You can find additional information in the [ASP.NET Core Documentation](https://docs.microsoft.com/en-us/aspnet/core/performance/objectpool).
```
Comment (BrennanConroy) [inline src/ObjectPool/Readme.md:3]:
```suggestion
This directory contains sources for [`Microsoft.Extensions.ObjectPool`](https://www.nuget.org/packages/Microsoft.Extensions.ObjectPool). This package provides types that enable object reuse. You can find additional information in the [ASP.NET Core Documentation](https://docs.microsoft.com/aspnet/core/performance/objectpool).
```

File: src/ObjectPool/Readme.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+## Description
+
+This project contains abstractions and default implementations for pooling objects. Commonly used types include:
+
+`ObjectPool<T>` - This represents a pool of objects. This is used to get and return pooled objects.
+`IPooledObjectPolicy<T>` = This policy defines how pooled objects are created and returned.
+`ObjectPoolProvider` - This represents a provider of `ObjectPool<T>`. This is used to create object pools based on an `IPooledObjectPolicy<T>`.
+
+
+For a full list of the types defined in this project, see [the namespace documentation](https://docs.microsoft.com/en-gb/dotnet/api/microsoft.extensions.objectpool).
```
Comment (BrennanConroy) [inline src/ObjectPool/Readme.md:14]:
```suggestion
For a full list of the types defined in this project, see [the namespace documentation](https://docs.microsoft.com/dotnet/api/microsoft.extensions.objectpool).
```

File: src/ObjectPool/Readme.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+## Development Setup
+
+### Build
+
+To build this specific project from source, follow the instructions [on building the project](../../docs/BuildFromSource.md#step-3:-build-the-repo).
```
Comment (BrennanConroy) [inline src/ObjectPool/Readme.md:20]:
```suggestion
To build this specific project from source, follow the instructions [on building the project](../../docs/BuildFromSource.md#step-3-build-the-repo).
```

File: src/ObjectPool/Readme.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
+# Object Pool
+
+This directory contains sources for [`Microsoft.Extensions.ObjectPool`](https://www.nuget.org/packages/Microsoft.Extensions.ObjectPool). This package provides types that enable object reuse. You can find additional information in the [ASP.NET Core Documentation](https://docs.microsoft.com/en-us/aspnet/core/performance/objectpool).
+
+## Description
+
+This project contains abstractions and default implementations for pooling objects. Commonly used types include:
+
+`ObjectPool<T>` - This represents a pool of objects. This is used to get and return pooled objects.
+`IPooledObjectPolicy<T>` = This policy defines how pooled objects are created and returned.
```
Comment (BrennanConroy) [inline src/ObjectPool/Readme.md:10]:
```suggestion
`IPooledObjectPolicy<T>` - This policy defines how pooled objects are created and returned.
```

**Review by BrennanConroy — APPROVED** [review]

File: src/Identity/README.md (hunk 4 above)
Comment (JunTaoLuo) [reply src/Identity/README.md:24]: Ah I didn't know that would work and was wondering what it does with special symbols. While both work, removing it is obviously better.

File: src/Analyzers/Readme.md (hunk 1 above)
Comment (captainsafia) [reply src/Analyzers/Readme.md]: Search and replace for this sounds good!

**Review by captainsafia — APPROVED** [review]

**Review by TanayParikh — APPROVED** [review]

File: src/Identity/README.md [hunk 9]
```diff
@@ -1,7 +1,35 @@
-ASP.NET Core Identity
-=====================
+# Identity
 
-ASP.NET Core Identity is the membership system for building ASP.NET Core web applications, including membership, login, and user data. ASP.NET Core Identity allows you to add login features to your application and makes it easy to customize data about the logged in user.
+ASP.NET Core Identity is the membership system for building ASP.NET Core web applications, including membership, login, and user data. ASP.NET Core Identity allows you to add login features to your application and makes it easy to customize data about the logged in user. You can find additional information in the [ASP.NET Core Documentation](https://docs.microsoft.com/aspnet/core/security/authentication/identity).
```
Comment (blowdart) [inline src/Identity/README.md:3]: Are we still saying ASP.NET _Core_ web applications?

Comment (Tratcher) [reply src/Identity/README.md:3]: *shrug* only .NET got rebranded, not ASP.NET Core.

**Pull request merged**